lib/Cleantalk/Antispam/EmailEncoder.php

```diff
@@ -268,38 +288,139 @@ protected function compileResponse($decoded_emails_array, $is_allowed)
      * Encoding any string
      *
      * @param $plain_string string
-     * @param $key string
      *
      * @return string
      */
-    private function encodeString($plain_string, $key)
+    public function encodeString($plain_string)
     {
-        if ( $this->encryption_is_available && !is_admin() ) {
-            $encoded_email = htmlspecialchars(@openssl_encrypt($plain_string, 'aes-128-cbc', $key));
+        if ( $this->use_ssl && $this->encryption_is_available ) {
+            $encoded_email = htmlspecialchars($this->openSSLEncrypt($plain_string));
         } else {
             $encoded_email = htmlspecialchars(base64_encode(str_rot13($plain_string)));
         }
-
         return $encoded_email;
     }
 
+    /**
+     * Encrypts a given plain string using the AES-128-CBC cipher algorithm and returns the encoded string.
+     *
+     * @param string $plain_string The plain text string that needs to be encrypted.
+     * @return string The encrypted string, which is a combination of the base64-encoded initialization vector (IV) and the encrypted data, separated by a predefined splitter.
+     */
+    private function openSSLEncrypt($plain_string)
+    {
+        global $apbct;
+        try {
+            if (!is_string($plain_string) || empty($plain_string)) {
+                throw new \Exception('Invalid or empty plain string');
+            }
+            // Determine the length of the IV required for the AES-128-CBC cipher algorithm
+            $iv_length = openssl_cipher_iv_length($this->cipher_algo);
+            if ($iv_length === false) {
+                throw new \Exception('Can\'t generate initializing vector length');
+            }
+            // Generate a random IV of the required length
+            $iv = openssl_random_pseudo_bytes($iv_length);
+            if (empty($iv)) {
+                throw new \Exception('Can\'t generate initializing vector body');
+            }
+            // Encrypt the plain string using the specified cipher algorithm, secret key, and IV
+            $encoded_string = @openssl_encrypt($plain_string, $this->cipher_algo, $this->secret_key, 0, $iv);
+            if (empty($encoded_string)) {
+                throw new \Exception('Can\'t encode plain string');
```

Comment:
Please, add more error info into the exception. Use the [`error_get_last()`](https://www.php.net/manual/en/function.error-get-last.php) function to get more details.